### PR TITLE
Adding bucket notification for archive zone for ObjectSync events

### PR DESCRIPTION
**Pull request by viduship**
Adding bucket notification for archive zone for ObjectSync events
This includes Object create and object delete at the active site, for which the notifications will happen for Create events but not for delete events at the archive zone.

Pass logs for bucket notification with archive zone :
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-QQCCFE

dependent cephci pr https://github.com/red-hat-storage/cephci/pull/3860

Ticket https://issues.redhat.com/browse/RHCEPHQE-14898

### CODE REVIEW

File: rgw/v2/tests/s3_swift/multisite_configs/test_bucket_notification_kafka_broker_archive_delete_replication_from_pri.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# test case CEPH-83575922
+config:
+ user_count: 1
+ bucket_count: 1
+ objects_count: 1
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/multisite_configs/test_bucket_notification_kafka_broker_archive_delete_replication_from_pri.yaml:5]: @viduship could you increase objects count to something like 25

File: rgw/v2/tests/s3_swift/reusables/bucket_notification.py [hunk 2]
```diff
@@ -234,12 +234,15 @@ def get_bucket_notification(rgw_s3_client, bucketname, empty=False):
     )
 
 
-def verify_event_record(event_type, bucket, event_record_path, ceph_version):
+def verify_event_record(event_type, bucket, event_record_path, ceph_version, config):
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/reusables/bucket_notification.py:237]: @viduship , verify_event_record is called in some other places also
so, could you make config=None here

File: rgw/v2/tests/s3_swift/reusables/bucket_notification.py [hunk 3]
```diff
@@ -234,12 +234,15 @@ def get_bucket_notification(rgw_s3_client, bucketname, empty=False):
...
     """
     verify event records
     """
     if os.path.getsize(event_record_path) == 0:
-        raise EventRecordDataError("event record not generated! File is empty")
+        if config.test_ops.get("test_delete_object_sync_archive", False):
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/reusables/bucket_notification.py:242]: @viduship as per the above change in method arguments, please modify this as below
if config and config.test_ops.get("test_delete_object_sync_archive", False):

File: rgw/v2/tests/s3_swift/reusables/bucket_notification.py [hunk 4]
```diff
@@ -369,9 +372,15 @@ def verify_event_record(event_type, bucket, event_record_path, ceph_version):
                 raise EventRecordDataError("awsRegion not in event record")
 
     if notifications_received is False:
-        raise EventRecordDataError(
-            f"Notifications not received for event type {event_type}"
-        )
+        if config.test_ops.get("test_delete_object_sync_archive", False):
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/reusables/bucket_notification.py:375]: @viduship as per the above change in method arguments, please modify this as below
if config and config.test_ops.get("test_delete_object_sync_archive", False):

File: rgw/v2/tests/s3_swift/reusables/bucket_notification.py (hunk 2 above)
Comment (viduship) [reply rgw/v2/tests/s3_swift/reusables/bucket_notification.py]: Done

File: rgw/v2/tests/s3_swift/reusables/bucket_notification.py (hunk 3 above)
Comment (viduship) [reply rgw/v2/tests/s3_swift/reusables/bucket_notification.py]: done

File: rgw/v2/tests/s3_swift/multisite_configs/test_bucket_notification_kafka_broker_archive_delete_replication_from_pri.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# test case CEPH-83575922
+config:
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/multisite_configs/test_bucket_notification_kafka_broker_archive_delete_replication_from_pri.yaml:2]: Please update the script name

File: rgw/v2/tests/s3_swift/reusables/bucket_notification.py [hunk 6]
```diff
@@ -234,12 +234,17 @@ def get_bucket_notification(rgw_s3_client, bucketname, empty=False):
     )
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/reusables/bucket_notification.py:234]: please update the config in the config section <input.yaml>

**Review by anrao19 — APPROVED** [review]
LGTM

**Review by hmaheswa — APPROVED** [review]

**Review by ckulal — APPROVED** [review]

**Pull request merged**